students/jstonham/README.md

```diff
@@ -2873,4 +2873,149 @@ Controller - Business logic.
+This allows components to be kept separate and updated independently of each other.
+
+Part of the reason that React is really performant is that change can only happen in one direction.
+Unlike other tools such as Angular and Ember, where change can flow from a component's state to the user *or* from the user to a component's state, React *only* allows change to flow from a component's state to the user.
+
+React components are made up of tags.
+There can only be one outermost tag in a React component, but other tags can be nested inside.
+In the following example, the `render()` method has an outermost `div` tag with `h1` and `ul` tags nested inside, and inside the `ul` tag are 3 `li` tags:
+
+```JSX
+render() {
+        return (
+            <div className="shopping-list">
+            <h1>ShoppingList for {this.props.name}</h1>
+            <ul>
+                <li>Instagram</li>
+                <li>WhatsApp</li>
+                <li>Oculus</li>
+            </ul>
+            </div>
+        );
+    }
+  ```
+
+## JSX (JavaScript Extended)
+
+JavaScript Extended is JavaScript with HTML.
+React can take in JSX and render as HTML.
+
+The following is a React component called `Greeting`, written in JSX, that will render the greeting message in HTML.
+
+```JSX
+class Greeting extends React.Component {
+    render() {
+        const name = this.props.name
+        return <p> Hi! My name is {name} </p>
+    }
+}
+```
+
+React components only need one method, `render()`, which will return the new component in a JSX format.
+
+Due to the fact that it's a combination of JavaScript and HTML, JSX needs to be able to differentiate between its JavaScript and HTML classes.
+The way it does so is to call its JavaScript classes `class` and its HTML classes `className`.
+
+## Data Structure
+
+Computer scientists think of data structures as trees (although it's easier to think of them as root systems).
+
+Example data structure:
+
+| | | Root (main Parent) | | | |
+| ---| --- | --- | --- | --- | --- |
+| | Parent 1 | | | Parent 2 | |
+| Child 1 | Child 2 | Child 3 | Child 1 | Child 2 | Child 3 |
+
+Parent components can have multiple child components, but child components can have only one parent component.
+
+This data structure is not only for React components, but for structures such as Lists and Maps.
+
+In React, the child components do not need to know their parent component.
+The parent component will dictate their props, but the child component does not need to know where their props come from.
+
+Browsers use a similar structure, called the Document Object Model (DOM) as a reference for placing elements on a web page.
+Elements are nested within each other in a tree-like structure and the locations of where the elements are placed are referred to as nodes.
+Medium has a good [article](https://medium.com/javascript-in-plain-english/how-to-traverse-the-dom-in-javascript-d6555c335b4e) that explains it all quite nicely.
```

Comment:
Just like GitHub isn't the author of your notes, Medium isn't the author of this article. Let's show our appreciation and attribute the article to the actual author instead of the platform here.